controllers/awsmachine_controller_test.go

```diff
@@ -1004,6 +1021,367 @@ func TestAWSMachineReconciler(t *testing.T) {
 	})
 }
 
+func TestAWSMachineReconciler_IntegrationTests(t *testing.T) {
+	var (
+		reconciler AWSMachineReconciler
+		mockCtrl   *gomock.Controller
+		recorder   *record.FakeRecorder
+	)
+
+	setup := func(t *testing.T, g *WithT) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		recorder = record.NewFakeRecorder(10)
+		reconciler = AWSMachineReconciler{
+			Client:   testEnv.Client,
+			Recorder: recorder,
+		}
+	}
+
+	teardown := func(g *WithT) {
+		mockCtrl.Finish()
+	}
+
+	t.Run("Should successfully reconcile control plane machine creation", func(t *testing.T) {
+		g := NewWithT(t)
+		mockCtrl = gomock.NewController(t)
+		ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+		secretMock := mock_services.NewMockSecretInterface(mockCtrl)
+		elbMock := mock_elbiface.NewMockELBAPI(mockCtrl)
+
+		expect := func(m *mock_ec2iface.MockEC2APIMockRecorder, s *mock_services.MockSecretInterfaceMockRecorder, e *mock_elbiface.MockELBAPIMockRecorder) {
+			mockedCreateInstanceCalls(m)
+			mockedCreateSecretCall(s)
+			mockedCreateLBCalls(e)
+		}
+		expect(ec2Mock.EXPECT(), secretMock.EXPECT(), elbMock.EXPECT())
+
+		ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("integ-test-%s", util.RandomString(5)))
+		g.Expect(err).To(BeNil())
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bootstrap-data",
+				Namespace: ns.Name,
+			},
+			Data: map[string][]byte{
+				"value": []byte("shell-script"),
+			},
+		}
+		g.Expect(testEnv.Create(ctx, secret)).To(Succeed())
+
+		setup(t, g)
+		awsMachine := getAWSMachine()
+		awsMachine.Namespace = ns.Name
+		createAWSMachine(g, awsMachine)
+
+		defer teardown(g)
+		defer t.Cleanup(func() {
+			g.Expect(testEnv.Cleanup(ctx, awsMachine, ns, secret)).To(Succeed())
+		})
+
+		cs, err := getClusterScope(infrav1.AWSCluster{ObjectMeta: metav1.ObjectMeta{Name: "test"}, Spec: infrav1.AWSClusterSpec{NetworkSpec: infrav1.NetworkSpec{Subnets: []infrav1.SubnetSpec{
+			{
+				ID:               "subnet-1",
+				AvailabilityZone: "us-east-1a",
+			}},
+		}}})
+		g.Expect(err).To(BeNil())
+		cs.Cluster = &clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "test-cluster"}}
```

Comment:
We should check if `cs !=nil`, otherwise `cs.Cluster` might break

Reply:
`cs` can never be nil, `cs.Cluster` also can never be nil as we are always creating it as part of `getClusterScope` func